docs/e2e-testing.md

```diff
@@ -0,0 +1,59 @@
+# E2E Testing  
+  
+## What is E2E Testing?
+  
+E2E (End-to-end) testing is a testing methodology where the objective is to test the application as a whole.   
+- It aims to ensure all integrated components of the application work together as expected when it is being used by the end user.   
+- This is done by simulating user scenarios on the fully built product.  
+  
+E2E tests in TEAMMATES can be found in the package `teammates.e2e`. 
+
+To set up and run E2E tests, refer to this [document](https://github.com/TEAMMATES/teammates/blob/master/docs/development.md#testing).  
+
+## Creating E2E tests
+  
+As E2E tests should be written from the end user perspective, each test case should reflect some user workflow.   
+  
+In TEAMMATES, E2E test cases are organized by page. Thus, for each page:  
+1. Identify the important user workflows  
+1. Simulate the user actions involved in the workflow by interacting with the UI elements.  
+1. Assert the expected conditions are present after the interaction.  
+  
+[Selenium](https://www.selenium.dev/) is used to locate and interact with elements in the UI.   
```

Comment:
Another important component for this E2E test is the `BackDoor` class. This should be mentioned somewhere.